tests/Integration/Behaviour/Features/Context/Domain/Product/ProductImageFeatureContext.php

```diff
@@ -52,6 +55,11 @@ class ProductImageFeatureContext extends AbstractProductFeatureContext
      */
     private $productImageRepository;
 
+    /**
+     * @var ShopProductImagesCollection
+     */
+    private $shopProductImagesCollection;
```

Comment:
With the changes you did on the step this stored collection is not useful anymore you can do it all in a single step:
```feature
  Scenario: Get image matrix
    When I try to get every image details for product "product1" in every shops I should get:
      #shop1
      | image reference |  cover | shopReference |
      | image1          |  1     | shop1         |
      | image2          |  0     | shop1         |
      #shop2
      | image1          |  0     | shop2         |
      | image2          |  0     | shop2         |
      | image3          |  0     | shop2         |
      #shop3
      | image1          |  0     | shop3         |
      | image2          |  0     | shop3         |
      | image3          |  0     | shop3         |
```

Reply:
I disagree. I Think we should stay on a Given->When-Then process. The way you suggest we mixed When and Then.
IMO it's less clear.

Reply:
Since we are going to add ShopProductImages with no images as well, we can't apply you're suggestion. Indeed to assert that some shops don't have any images we can't do that in only one step.